Explicitly document how to template manifests with both `helm` and `istioctl` and remove references to `istioctl manifest diff` (#16057)
istio/istio.io: content/en/docs/setup/install/istioctl/index.md

    $ istioctl manifest generate --set values.defaultRevision=default
    {{< /text >}}

1. Resources may not be installed with the same sequencing of dependencies as

> `istioctl install` and is not tested as part of Istio releases.

Reviewer: ```suggestion
`istioctl install` and may have subtle differences from the manifests tested as part of Istio releases.
```

that you follow [these steps](/docs/ops/best-practices/security/#configure-third-party-service-account-tokens) if your
Kubernetes environment does not support third party service account tokens.

> Note that it is recommended to append `--cluster-specific` to your `istio manifest generate` command to detect the target cluster's environment, which will embed those cluster-specific environment settings into the generated manifests. This does require network access to a live cluster, however.

Reviewer: ```suggestion
Note that it is recommended to append `--cluster-specific` to your `istio manifest generate` command to detect the target cluster's environment, which will embed those cluster-specific environment settings into the generated manifests. This requires network access to your running cluster.
```
